receiver/prometheusreceiver/metrics_receiver.go

```diff
@@ -138,6 +139,7 @@ func (pr *Preceiver) StartMetricsReception(host receiver.Host) error {
 		ctx := host.Context()
 		c, cancel := context.WithCancel(ctx)
 		pr.cancel = cancel
+		c = observability.ContextWithReceiverName(c, observability.MakeComponentName(typeStr, pr.MetricsSource()))
```

Comment:
This should be config.Type() and Config.Name() (the whole config not the Prometheus special config)